ICA/auto_mycosnp_ICA.py

```diff
@@ -0,0 +1,187 @@
+    runs_completed_analysis=os.listdir("/Volumes/IDGenomics_NAS/ICA/Runs_On_ICA")
+#This maybe unneeded but during testing I think the os package generates this file that needs to be removed from list
+    try:
+        runs_completed_analysis.remove('.DS_Store')
+    except:
+        print("Nope")
+    try:
+        runs_completed_analysis.remove('@eaDir')
+    except:
+        print("Nope")
+
+#This checks if an anaylsis needs to be kicked off if not the script sleeps for an hour
+    if len(runs_completed_analysis) == len(runs_on_ica):
+        print('No new runs waiting 1 hr')
+        time.sleep(3600)
+
+
+    else:
+        #Formats string to work in ICAv2
+        run=list(set(runs_on_ica)-set(runs_completed_analysis))
+        run='/datasets/'+run[0]+'/'
+
+        try:
+            os.remove('./bsshinput.json')
+        except:
+            print("No ./bsshinput.json")
+
+        # The bsshinput.json has our experiment name that we typically use i.e. UT-VH0770-220318
+        bashCommand = "icav2 projectdata list --data-type FILE --parent-folder %s --file-name bsshinput.json" % (run)
+        print(bashCommand)
+        tmp= icav_out(bashCommand)
+        download=tmp[0].split(' ')[4]
+
+        bashCommand = "icav2 projectdata download %s" % download
+        process = subprocess.Popen(bashCommand.split(), stdout=subprocess.PIPE)
+        process.communicate()
+
+        Experiment=json.load(open('./bsshinput.json'))['BsshRun']['ExperimentName']
+
+    #Mycosnp needs a samplesheet, as input and must be uploaded into ICA
+        try:
+            os.mkdir("./%s_samplesheet/" % Experiment)
+        except:
+            print("Folder Already Exits")
+
+    #The names of the samples, and there ICA ids must be downloaded from ICA
+        bashCommand = "icav2 projectdata list  --parent-folder " +  run + "output/fastqs/Samples/"
+        tmp= icav_out(bashCommand)
+
+        ica_lanes=[]
+        for i in tmp:
+            ica_lanes.append(i.split(" ")[1])
+
+        ica_names=[]
+        for i in ica_lanes:
+            bashCommand = "icav2 projectdata list  --parent-folder  " +  run + "output/fastqs/Samples/Lane_1/"
+            tmp= icav_out(bashCommand)
+            for j in tmp:
+                if not "Undetermined" in j.split(" ")[0]:
+                    ica_names.append(j.split(" ")[0])
+
+        ica_ids=[]
+        sample_names=[]
+        for i in ica_names:
+            bashCommand = 'icav2 projectdata list  --parent-folder ' + i
+            tmp= icav_out(bashCommand)
+            for j in tmp:
+                ica_ids.append(j.split(" ")[4]+",")
+                sample_names.append(j.split(" ")[1])
+
+
+        sample_sheet=[]
+        sample_sheet.append('sample_id,fastq_1,fastq_2\n')
+        for i in sample_names:
+            if i.split('_')[3] == 'R1':
+                tmp=i.split('_')[0]+','+i
+                tmp=tmp+','+i.replace('R1','R2')+'\n'
+                sample_sheet.append(tmp)
```

Comment:
We need the sample sheet creation be a separate script so it can be externally without the loop